### PR TITLE
support aarch64 for raspberrypi3b

### PR DESCRIPTION
**Pull request by churmenton**
**- What I did**
modified the machine configuration file of raspberrypi3 and copy raspberrypi3-64.conf from morty branch
changed the Linux kernel
modified the device tree variant
**- How I did it**
Reference to Morty for the supporting of aarch64

### CODE REVIEW

**Comment by pbrkr** [comment]
I wouldn't like to make an invasive change like this on an old branch as it risks breaking things for users stuck on this version.

**Comment by agherzan** [comment]
@churmenton Please backport specific commits with cherrypick.
@pbrkr That is a fair point. What do you reckon to do?

**Comment by agherzan** [comment]
Also, @churmenton, please checkout https://github.com/agherzan/meta-raspberrypi/blob/master/docs/contributing.md

**Comment by pbrkr** [comment]
I'd recommend either doing this in a way that doesn't affect `raspberrypi3` (32-bit) at all (which would involve adding a separate recipe for the new firmware version and setting `PREFERRED_PROVIDER` or `PREFERRED_VERSION` values as appropriate), or simply maintaining a fork of this branch. It's unlikely that we're going to make many changes to old branches like this so a fork isn't going to be expensive for @churmenton to maintain.

File: conf/machine/include/rpi-base.inc [hunk 1]
```diff
@@ -2,6 +2,10 @@ include conf/machine/include/rpi-default-settings.inc
 include conf/machine/include/rpi-default-versions.inc
 include conf/machine/include/rpi-default-providers.inc
 
+require conf/multilib.conf
+MULTILIBS = "multilib:lib32"
+DEFAULTTUNE_virtclass-multilib-lib32 = "armv7athf-neon-vfpv4"
+
```
Comment (pbrkr) [inline conf/machine/include/rpi-base.inc]: This bit doesn't look like a backport - I can't see it in the newer branches.

File: recipes-kernel/linux/linux-rpi.inc [hunk 2]
```diff
@@ -13,9 +13,9 @@ ARM_KEEP_OABI ?= "1"
 # Quirk for udev greater or equal 141
 UDEV_GE_141 ?= "1"
 
-# Set the verbosity of kernel messages during runtime
-# You can define CMDLINE_DEBUG in your local.conf or distro.conf to override this behaviour
-CMDLINE_DEBUG ?= '${@base_conditional("DISTRO_TYPE", "release", "quiet", "debug", d)}'
+# You can define CMDLINE_DEBUG as "debug" in your local.conf or distro.conf
+# to enable kernel debugging.
+CMDLINE_DEBUG ?= ""
```
Comment (pbrkr) [inline recipes-kernel/linux/linux-rpi.inc]: This change is unrelated to adding `raspberrypi3-64` support.

File: conf/machine/include/rpi-base.inc (hunk 1 above)
Comment (churmenton) [reply conf/machine/include/rpi-base.inc]: @pbrkr This is used for supporting 32bit lib on 64bit platform. When some 3rd party binary is delivered on a 32bit form, we can still run it on raspberrypi3-64 version.

File: recipes-kernel/linux/linux-rpi.inc (hunk 2 above)
Comment (churmenton) [reply recipes-kernel/linux/linux-rpi.inc]: @pbrkr I just copied it from Morty branch^_^

File: conf/machine/include/rpi-base.inc (hunk 1 above)
Comment (pbrkr) [reply conf/machine/include/rpi-base.inc]: We shouldn't be adding features to old branches which haven't been merged to master as it fragments our support. It'd be good to see this proposed for master though.

**Comment by churmenton** [comment]
> Also, @churmenton, please checkout https://github.com/agherzan/meta-raspberrypi/blob/master/docs/contributing.md

Thanks for your remind, I will read this material.

**Comment by churmenton** [comment]
> @churmenton Please backport specific commits with cherrypick.
OK, I will do it later.

File: conf/machine/include/rpi-base.inc (hunk 1 above)
Comment (kraj) [reply conf/machine/include/rpi-base.inc]: yes enabling multilib is a distro choice not BSP choice. Please remove this

Comment (churmenton) [reply conf/machine/include/rpi-base.inc]: @kraj OK, I will remove this snippets
